pkg/driver/controller.go

```diff
@@ -222,6 +222,9 @@ func (d *ControllerService) CreateVolume(ctx context.Context, req *csi.CreateVol
 	if modifyOptions.modifyDiskOptions.Throughput != 0 {
 		throughput = modifyOptions.modifyDiskOptions.Throughput
 	}
+	if modifyOptions.modifyTagsOptions.TagsToAdd != nil {
+		d.options.ExtraTags = modifyOptions.modifyTagsOptions.TagsToAdd
```

Comment:
If you overwrite `d.options.ExtraTags`, any tags from the CLI won't be added:
```suggestion
	for k, v := range modifyOptions.modifyTagsOptions.TagsToAdd {
	  volumeTags[k] = v
	}
```

Also, this needs to be after https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/d1d79a81397b39302ae4145559fd10e6c49b8e92/pkg/driver/controller.go#L313-L315 because mutable parameters should take precedence over non-mutable parameters according to the CSI spec

Reply:
Thanks for pointing this out, noticed after unit test failed, resolved!